[docs] Add Expo Orbit page (#25855)
expo/expo: docs/pages/build/orbit.mdx

---
title: Expo Orbit
maxHeadingDepth: 4
description: Accelerate your development workflow with one-click build launches and simulator management.
---

import { Terminal } from '~/ui/components/Snippet';
import Video from '~/components/plugins/Video';

> Expo Orbit for macOS makes it faster to install and run builds from EAS or run Snack projects, on simulators and physical devices.

Author: ```suggestion
Expo Orbit for macOS makes it faster to install and run builds from EAS, local files, or run Snack projects, on simulators and physical devices.
```

> Expo Orbit for macOS makes it faster and easier to install and run builds from EAS or elsewhere, and to run Snack projects on simulators and physical devices.

Reviewer: ```suggestion
Expo Orbit for macOS makes it faster to install and run builds from EAS or run Snack projects, on simulators and physical devices.
```

- What does elsewhere here mean? Since we are introducing Orbit for the first time in our docs, let's be as explicit as we can about the use cases.
- Going to remove the phrase "easier" as we can leave that for our user base to decide. I think using the phrase "faster ..." is much more useful.
- Let's also add a pause before sims and physical device so that it indicates that both EAS builds and snacks can be used here.

Reviewer: This sounds great! Thanks for the context. Yeah, having this makes it clear.

<Video file="orbit/basic-features.mp4" />

> The process for installing builds from EAS (Android device/emulator, iOS simulator/device) and running Snack projects on simulators used to be more manual than it needs to be. You could use `eas build:run` and select a build for Android devices/emulators or iOS simulators, or you could download the archive, extract it, and drag and drop it to the simulator/emulator. For Snack projects, additional steps include installing Expo Go in the simulator through the CLI, logging in, and selecting the Snack from a list. Orbit makes these steps as seamless as possible.

Reviewer: ```suggestion
Before Orbit, installing builds from EAS (Android and iOS physical devices or emulator/simulator) or running Snack projects on simulators was manual. You had to run `eas build:run` command and select a build for your chosen device or download the archive and then drag and drop it to the simulator (in the case of iOS). Also, for Snack projects, additional steps included installing Expo Go on the virtual device, logging in, and then selecting the Snack from the list. Orbit makes all of these steps as seamless as possible. 
```

If we are providing the historical context, and use past tense, let's make sure to use the past tense through the paragraph. I also tried to make it a bit more concise, repeating emulator/simulator too many times was cumbersome to read.

> The process for installing builds from EAS (Android device/emulator, iOS simulator, iOS device) and running Snack projects on simulators used to be more manual than it needs to be. You could use `eas build:run` and select a build for Android devices/emulators or iOS simulators, or you could download the archive, extract it, and drag and drop it to the simulator/emulator. For Snack projects, additional steps include installing Expo Go in the simulator through the CLI, logging in, and selecting the Snack from a list. To improve this we released Orbit, making these steps as seamless as possible and aligning with the user-friendly experience that Expo offers

Reviewer 2: ```suggestion
The process for installing builds from EAS (Android device/emulator, iOS simulator/device) and running Snack projects on simulators used to be more manual than it needs to be. You could use `eas build:run` and select a build for Android devices/emulators or iOS simulators, or you could download the archive, extract it, and drag and drop it to the simulator/emulator. For Snack projects, additional steps include installing Expo Go in the simulator through the CLI, logging in, and selecting the Snack from a list. Orbit makes these steps as seamless as possible.
```

## Highlights

- List and launch simulators, including the ability to run Android emulators without audio

> - Install and launch builds from EAS to your simulators and real devices with just one click

Reviewer 2: ```suggestion
- Install and launch builds from EAS to your simulators and real devices in one click
```

> - Launch Snack projects in your simulators by clicking a button

Reviewer 2: ```suggestion
- Launch Snack projects in your simulators in one click
```

> - List and launch simulators, including the ability to run Android emulators without audio
> - Install and launch builds from EAS to your simulators and real devices in one click
> - Launch Snack projects in your simulators in one click
> - Install and launch apps from local files using Finder or just drag and drop a file into the menu bar app. Orbit supports any Android APK, simulator compatible **.app**, or ad hoc signed apps
> - See pinned projects and quickly launch your latest builds

Reviewer: ```suggestion
- List and launch simulators, including running Android emulators without audio.
- Install and launch builds from EAS to your simulators and real devices in one click.
- Launch Snack projects in your simulators in one click.
- Install and launch apps from local files using Finder or drag and drop a file into the menu bar app. Orbit supports any Android **.apk**, iOS simulator compatible **.app**, or ad hoc signed apps.
- See pinned projects from your EAS dashboard and quickly launch your latest builds.
```

> - Install and launch apps from local files using Finder or just drag and drop a file into the menu bar app. Orbit supports any Android apk, simulator compatible .app, or ad-hoc signed apps

Reviewer 2: ```suggestion
- Install and launch apps from local files using Finder or just drag and drop a file into the menu bar app. Orbit supports any Android APK, simulator compatible **.app**, or ad hoc signed apps
```

canonical spelling for ad-hoc as per apple is ad hoc - https://developer.apple.com/help/account/manage-profiles/create-an-ad-hoc-provisioning-profile/

Author: makes sense, thanks for fixing this

## Installation

> You can download Orbit via Homebrew or directly through [GitHub](https://github.com/expo/orbit/releases).

Reviewer 3: ```suggestion
You can download Orbit with Homebrew, or directly from the [GitHub releases](https://github.com/expo/orbit/releases).
```

Phrasing nit.

<Terminal cmd={['$ brew install expo-orbit']} />

> If you want Orbit to automatically start when you log in, click on the Orbit icon in the menu bar, then "Settings" and select the "Launch on Login" option.

Reviewer: ```suggestion
If you want Orbit to start when you log in automatically, click on the Orbit icon in the menu bar, then **Settings** and select the **Launch on Login** option.
```

Let's make the settings name bold.

> > **info** Orbit relies on the Android SDK and `xcrun` for device management, which requires having both Android Studio and Xcode set up.

Reviewer: ```suggestion
> **info** Orbit relies on the Android SDK and `xcrun` for device management, which requires setting up both [Android Studio](/workflow/android-studio-emulator/) and [Xcode](/workflow/ios-simulator/).
```

> At this point in time, Orbit is compatible only with macOS but we have exciting plans to integrate it further into the Expo ecosystem and add even more features. Try out Expo Orbit now, explore its capabilities, and share your feedback.

Reviewer: ```suggestion
Right now, Orbit is compatible only with macOS. However, we have exciting plans to integrate it further into the Expo ecosystem and add even more features. Try out Expo Orbit now, explore its capabilities, and share your feedback.
```
